### ISSUE TITLE
Hang while running proxy unit tests on Go 1.5

### ISSUE THREAD

**Issue by stevvooe** [docker/distribution#913]
I am getting a hang while running the unit tests in Go 1.5 with `make test`. I cannot seem to reproduce this issue by running just `go test` in the package.

```console
$ make test
+ test
?   	github.com/docker/distribution	[no test files]
?   	github.com/docker/distribution/cmd/registry	[no test files]
?   	github.com/docker/distribution/cmd/registry-api-descriptor-template	[no test files]
ok  	github.com/docker/distribution/configuration	0.052s
ok  	github.com/docker/distribution/context	2.037s
ok  	github.com/docker/distribution/digest	3.214s
ok  	github.com/docker/distribution/health	0.013s
ok  	github.com/docker/distribution/health/api	0.014s
ok  	github.com/docker/distribution/health/checks	0.616s
?   	github.com/docker/distribution/manifest	[no test files]
ok  	github.com/docker/distribution/manifest/schema1	0.132s
ok  	github.com/docker/distribution/notifications	9.915s
?   	github.com/docker/distribution/registry	[no test files]
ok  	github.com/docker/distribution/registry/api/errcode	0.047s
ok  	github.com/docker/distribution/registry/api/v2	0.064s
?   	github.com/docker/distribution/registry/auth	[no test files]
ok  	github.com/docker/distribution/registry/auth/htpasswd	0.043s
ok  	github.com/docker/distribution/registry/auth/silly	0.019s
ok  	github.com/docker/distribution/registry/auth/token	0.209s
ok  	github.com/docker/distribution/registry/client	0.838s
ok  	github.com/docker/distribution/registry/client/auth	0.093s
?   	github.com/docker/distribution/registry/client/transport	[no test files]
ok  	github.com/docker/distribution/registry/handlers	82.518s
?   	github.com/docker/distribution/registry/listener	[no test files]
?   	github.com/docker/distribution/registry/middleware/registry	[no test files]
?   	github.com/docker/distribution/registry/middleware/repository	[no test files]
SIGQUIT: quit
PC=0x5f39b m=0

goroutine 0 [idle]:
runtime.mach_semaphore_wait(0xd03, 0x0, 0x1f3fc, 0x67, 0x0, 0x70ce00, 0x51f39, 0xffffffffffffffff, 0x1cd20, 0x7fff5fbff61c, ...)
	/usr/local/go/src/runtime/sys_darwin_amd64.s:407 +0xb
runtime.semasleep1(0xffffffffffffffff, 0x1cd20)
	/usr/local/go/src/runtime/os1_darwin.go:385 +0xe5
runtime.semasleep.func1()
	/usr/local/go/src/runtime/os1_darwin.go:401 +0x29
runtime.systemstack(0x7fff5fbff620)
	/usr/local/go/src/runtime/asm_amd64.s:278 +0xab
runtime.semasleep(0xffffffffffffffff, 0x0)
	/usr/local/go/src/runtime/os1_darwin.go:402 +0x36
runtime.notesleep(0x70d4b0)
	/usr/local/go/src/runtime/lock_sema.go:169 +0x100
runtime.stopm()
	/usr/local/go/src/runtime/proc1.go:1128 +0x112
runtime.findrunnable(0xc820021500, 0x0)
	/usr/local/go/src/runtime/proc1.go:1530 +0x69e
runtime.schedule()
	/usr/local/go/src/runtime/proc1.go:1639 +0x267
runtime.park_m(0xc820a94480)
	/usr/local/go/src/runtime/proc1.go:1698 +0x18b
runtime.mcall(0x7fff5fbff7a0)
	/usr/local/go/src/runtime/asm_amd64.s:204 +0x5b

goroutine 1 [chan receive, 9 minutes]:
testing.RunTests(0x593960, 0x7010e0, 0x3, 0x3, 0x1)
	/usr/local/go/src/testing/testing.go:562 +0x8ad
testing.(*M).Run(0xc82005fef8, 0xc820137a40)
	/usr/local/go/src/testing/testing.go:494 +0x70
main.main()
	github.com/docker/distribution/registry/proxy/_test/_testmain.go:58 +0x116

goroutine 17 [syscall, 9 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 6 [runnable]:
io.(*SectionReader).Read(0xc82016c1e0, 0xc820a96000, 0x8000, 0x8012, 0x0, 0xaa0028, 0xc820012180)
	/usr/local/go/src/io/io.go:447
io/ioutil.(*nopCloser).Read(0xc82016e360, 0xc820a96000, 0x8000, 0x8012, 0x30df, 0x0, 0x0)
	<autogenerated>:4 +0x82
io.(*teeReader).Read(0xc820168720, 0xc820a96000, 0x8000, 0x8012, 0x1, 0x0, 0x0)
	/usr/local/go/src/io/io.go:514 +0x7c
github.com/docker/distribution/registry/proxy.streamToClient(0xaa4c90, 0x70c760, 0xaa5510, 0xc820149e40, 0x522e80, 0x18, 0x5, 0xc820015130, 0x47, 0xd00000, ...)
	/Users/sday/go/src/github.com/docker/distribution/registry/proxy/proxyblobstore.go:165 +0x319
github.com/docker/distribution/registry/proxy.proxyBlobStore.ServeBlob(0xaa50a8, 0xc8201496e0, 0xaa5058, 0xc8201496c0, 0xc82001ad80, 0xaa4c90, 0x70c760, 0xaa5510, 0xc820149e40, 0xc8200a41c0, ...)
	/Users/sday/go/src/github.com/docker/distribution/registry/proxy/proxyblobstore.go:81 +0x62d
github.com/docker/distribution/registry/proxy.TestProxyStoreServe(0xc8200866c0)
	/Users/sday/go/src/github.com/docker/distribution/registry/proxy/proxyblobstore_test.go:183 +0x579
testing.tRunner(0xc8200866c0, 0x7010f8)
	/usr/local/go/src/testing/testing.go:456 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:561 +0x86d

rax    0xe
rbx    0xd03
rcx    0x7fff5fbff5a8
rdx    0x7fff5fbff620
rdi    0xd03
rsi    0x0
rbp    0x70d3a0
rsp    0x7fff5fbff5a8
r8     0x7fff5fbff560
r9     0x7fff5fbff1f4
r10    0x0
r11    0x286
r12    0x2e43199648ef3
r13    0x2ecac2ab18788
r14    0x13fc6091c26a8000
r15    0x8
rip    0x5f39b
rflags 0x286
cs     0x7
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (10m0s).
FAIL	github.com/docker/distribution/registry/proxy	600.025s
ok  	github.com/docker/distribution/registry/proxy/scheduler	0.757s
ok  	github.com/docker/distribution/registry/storage	44.152s
?   	github.com/docker/distribution/registry/storage/cache	[no test files]
ok  	github.com/docker/distribution/registry/storage/cache/memory	0.141s
ok  	github.com/docker/distribution/registry/storage/cache/redis	0.169s
?   	github.com/docker/distribution/registry/storage/driver	[no test files]
ok  	github.com/docker/distribution/registry/storage/driver/azure	0.112s
?   	github.com/docker/distribution/registry/storage/driver/base	[no test files]
?   	github.com/docker/distribution/registry/storage/driver/factory	[no test files]
ok  	github.com/docker/distribution/registry/storage/driver/filesystem	83.822s
ok  	github.com/docker/distribution/registry/storage/driver/inmemory	81.675s
?   	github.com/docker/distribution/registry/storage/driver/middleware	[no test files]
?   	github.com/docker/distribution/registry/storage/driver/middleware/cloudfront	[no test files]
?   	github.com/docker/distribution/registry/storage/driver/oss	[no test files]
?   	github.com/docker/distribution/registry/storage/driver/rados	[no test files]
ok  	github.com/docker/distribution/registry/storage/driver/s3	0.082s
ok  	github.com/docker/distribution/registry/storage/driver/swift	80.885s
?   	github.com/docker/distribution/registry/storage/driver/testsuites	[no test files]
?   	github.com/docker/distribution/testutil	[no test files]
ok  	github.com/docker/distribution/uuid	0.114s
?   	github.com/docker/distribution/version	[no test files]
make: *** [test] Error 1
```

**Comment by stevvooe** [comment]
@RichardScothern PTAL

**Comment by RichardScothern** [comment]
I see this on the build machine with 1.5, on my laptop with 1.5-rc1, but not on my laptop with 1.5.

Investigating..

**Comment by RichardScothern** [comment]
Fixed by #1006
